herald/herald/Sensor/BLE/BLEReceiver.swift

```diff
@@ -775,8 +776,8 @@ class ConcreteBLEReceiver: NSObject, BLEReceiver, BLEDatabaseDelegate, CBCentral
         }
         logger.debug("didDiscover (device=\(device),rssi=\((String(describing: device.rssi))),txPower=\((String(describing: device.txPower))))")
         // Process legacy advert only protocol
-        let legacyAdvertOnlyProtocolData = (BLESensorConfiguration.legacyAdvertOnlyProtocolServiceUUID == nil ? nil : BLELegacyAdvertOnlyProtocolData(fromAdvertisementData: advertisementData))
-        if let legacyAdvertOnlyProtocolData = legacyAdvertOnlyProtocolData {
+        let legacyAdvertOnlyProtocolData = (BLESensorConfiguration.interopAdvertBasedProtocolEnabled ? BLELegacyAdvertOnlyProtocolData(fromAdvertisementData: advertisementData) : nil)
```

Comment:
Does this line throw if an app developer sets enabled to true but does not provide the config for where to find the pseudo address?

Reply:
Well spotted. This was a potential issue in previous versions of the code.
- All interop configuration values are not optionals. Default values are provided.
- BLELegacyAdvertOnlyProtocolData(fromAdvertisementData: advertisementData) also has error checks.
- Have double checked, won't throw.
- Updated all configuration values to make optionals and non-optionals explicit in declaration.